web/frontend/src/api/sensors.ts

```diff
@@ -0,0 +1,17 @@
+import { grpc } from '@improbable-eng/grpc-web';
+import { sensorsApi, type ResourceName, Client } from '@viamrobotics/sdk';
+
+export const getSensors = (client: Client, name: string) => {
+  const request = new sensorsApi.GetSensorsRequest();
+  request.setName(name);
+
+  return new Promise<ResourceName[]>((resolve, reject) => {
+    client.sensorsService.getSensors(request, new grpc.Metadata(), (error, response) => {
+      if (error) {
+        reject(error);
+      }
+
+      resolve((response?.toObject().sensorNamesList) ?? []);
```

Comment:
- `resolve` is not guarded by `else` or an early return
    - `new Promise` will throw away a `resolve` after a `reject`, but I think it's better to be explicit
- Move response mapping to `then` block

Reply:
Updated!